packages/e2e.sandbox.subscribeapp/src/MapComponent.tsx

```diff
@@ -0,0 +1,177 @@
+
+export interface Coordinates {
+  lat: number;
+  lng: number;
+}
+
+export interface EntityInformation {
+  id: string;
+  routeId: string;
+  nextStopId: string;
+}
+
+interface EntityInformationMap {
+  [key: string]: EntityInformation;
+}
+interface ObjectCoordinatesMap {
+  [key: string]: Coordinates;
+}
+
+const MapComponent: React.FC = () => {
+  const center: Coordinates = { lat: 40.7128, lng: -74.0060 };
+
+  const [entityInfo, setEntityInfo] = React.useState<EntityInformationMap>({});
+
+  const [objectCoordinates, setObjectCoordinates] = React.useState<
+    ObjectCoordinatesMap
+  >({});
+
+  const initializeEntity = (info: EntityInformation) => {
+    setEntityInfo((prev) => ({
+      ...prev,
+      [info.id]: info,
+    }));
+  };
+
+  const deleteEntity = (info: EntityInformation) => {
+    setEntityInfo((prev) => {
+      const newEntityInfo = { ...prev };
+      delete newEntityInfo[info.id];
+      return newEntityInfo;
+    });
+  };
+
+  const updateCoordinates = (primaryKey: string, coordinates: Coordinates) => {
+    if (!(primaryKey in entityInfo)) {
+      return;
+    }
+    setObjectCoordinates((prev) => ({
+      ...prev,
+      [primaryKey]: coordinates,
+    }));
+  };
+
+  const fetchInitialPoints = async () => {
+    const page = await foundryClient($Objects.MtaBus).fetchPageWithErrors({
+      $pageSize: 50,
+    });
+    if (isOk(page)) {
+      const objects = page.value.data;
+
+      for (const object of objects) {
+        const primaryKey = object.$primaryKey;
+        initializeEntity({
+          id: primaryKey,
+          routeId: object.routeId
+            ?? "",
+          nextStopId: object.nextStopId ?? "",
+        });
+      }
+      for (const object of objects) {
+        const primaryKey = object.$primaryKey;
+        object.positionId?.getLatestValue().then(
+          (latestPosition) => {
+            console.log("Latest position is", latestPosition);
+            if (latestPosition != null) {
+              const newCoordinates = {
+                lat: latestPosition?.value.coordinates[1]!,
+                lng: latestPosition?.value.coordinates[0]!,
+              };
+              updateCoordinates(primaryKey, newCoordinates);
+            }
+          },
+        );
+      }
+    }
+  };
+
+  React.useEffect(() => {
+    fetchInitialPoints();
+  }, []);
+
+  foundryClient($Objects.MtaBus)[__EXPERIMENTAL__NOT_SUPPORTED_YET_subscribe](
+    ["nextStopId", "positionId", "routeId", "vehicleId"],
+    {
+      onChange: (object) => {
+        const primaryKey = object.object.$primaryKey;
+
+        if (!(primaryKey in entityInfo)) {
+          return;
+        }
+
+        console.log(
+          "Received update for",
+          primaryKey,
+          ". Length of map is ",
+          Object.keys(entityInfo).length,
+        );
+
+        if (object.state === "REMOVED") {
+          deleteEntity(entityInfo[primaryKey]);
+        }
+
+        if (object.object.positionId != null) {
```

Comment:
Is this actually something more like event.bus.positionId in context?

Reply:
yes, I should enforce better naming conventions since it's a sample app